engine.py

```diff
@@ -694,21 +692,24 @@ def show_dialog(self, title, *args, **kwargs):
 
         :returns: the created widget_class instance
         """
-        if not sgtk.util.is_macos():
-            return super(MayaEngine, self).show_dialog(title, *args, **kwargs)
-        else:
-            if not self.has_ui:
-                self.log_error(
-                    "Sorry, this environment does not support UI display! Cannot show "
-                    "the requested window '%s'." % title
-                )
-                return None
+        from sgtk.platform.qt import QtCore, QtGui
 
-            from sgtk.platform.qt import QtCore, QtGui
+        if not self.has_ui:
+            self.log_error(
+                "Sorry, this environment does not support UI display! Cannot show "
+                "the requested window '%s'." % title
+            )
+            return None
```

Comment:
I think you should keep the "import Qt" line after the "if has UI" block like in the original code. Because if "no UI" is true, that means the import will probably fail so Python will crash instead of giving this nice log message.

Reply:
Good catch! thanks.